### PR TITLE
Matching Vercel's requirements for sponsorship

### PR DESCRIPTION
**Pull request by marehr**
This PR contains build infrastructure changes to build our documentation on vercel.com. This will allow us to have previews of our documentation. Part of https://github.com/seqan/product_backlog/issues/198

Their stated requirement were

* [x] The project should be non-commercial.
* [x] The project should be open source.
* [x] The project should be static rather than server-rendered.
* [x] The project should not be related in any way to cryptocurrency.
* [x] A Vercel banner in the footer of each site page.
* [x] A Vercel banner in the source repository's `README.md` file.
* [x] All links back to Vercel should use a UTM tag with the format: `?utm_source=[team-name]&utm_campaign=oss`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,3 +68,8 @@ g++-7 -O3 -DNDEBUG -Wall -Wextra                                \
     -lz -lbz2 -lstdc++fs -pthread                               \
   your_file.cpp
 ```
+
+## Sponsorships
+
+* [![Vercel](https://raw.githubusercontent.com/seqan/seqan3/master/test/documentation/.vercel/powered-by-vercel.svg)](https://vercel.com/?utm_source=seqan&utm_campaign=oss):
+  Vercel is kindly enough to sponsor our documentation preview-builds within our pull requests. Check them out!
```
Comment (eseiler) [inline README.md:75]:
```suggestion
  Vercel is kind enough to sponsor our documentation preview-builds within our pull requests. Check them out!
```

File: test/documentation/.vercel/install.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -x
+set -e
+
+DOXYGEN_VERSION=1.9.1
```
Comment (eseiler) [inline test/documentation/.vercel/install.sh:5]: Doesn't have to be now, but I am wondering if there could be a way to generalize this...
We are setting the doxygen version in 2 github action workflows and now here...

Comment (marehr) [reply test/documentation/.vercel/install.sh:5]: Yep, I think we should parse the content from github actions.

File: test/documentation/seqan3-doxygen.cmake [hunk 3]
```diff
@@ -24,10 +24,19 @@ else ()
...
+set (SEQAN3_DOXYGEN_SOURCE_DIR "${SEQAN3_CLONE_DIR}")
+set (SEQAN3_DOXYFILE_IN ${SEQAN3_DOXYGEN_INPUT_DIR}/seqan3_doxygen_cfg.in)
+set (SEQAN3_FOOTER_HTML_IN ${SEQAN3_DOXYGEN_INPUT_DIR}/seqan3_footer.html.in)
 
 option(SEQAN3_USER_DOC "Create build target and test for user documentation." ON)
 option(SEQAN3_DEV_DOC "Create build target and test for developer documentation." ON)
+option(SEQAN3_VERCEL_PREVIEW_DOC "Is this a preview build by vercel.com?" OFF)
+
+if (SEQAN3_VERCEL_PREVIEW_DOC)
+    set (SEQAN3_DOXYFILE_OPTION_POWERED_BY_VERCEL "HTML_EXTRA_FILES       += ${SEQAN3_DOXYGEN_SOURCE_DIR}/test/documentation/.vercel/powered-by-vercel.svg")
```
Comment (Irallia) [inline test/documentation/seqan3-doxygen.cmake:36]: Why are there these empty spaces?

**Review by Irallia — APPROVED** [review]
LGFM, just a question about some spaces.

File: test/documentation/seqan3-doxygen.cmake (hunk 3 above)
Comment (eseiler) [reply test/documentation/seqan3-doxygen.cmake:36]: For alignment, when it gets appended:

https://github.com/seqan/seqan3/blob/0721fec7db931ce7d5588982c60588d159bed6a8/test/documentation/seqan3_doxygen_cfg.in#L106

Comment (marehr) [reply test/documentation/seqan3-doxygen.cmake:36]: Copy paste error :)

**Review by eseiler — APPROVED** [review]

**Pull request merged**